docs/introduction/migration-v4-v5.md

```diff
@@ -0,0 +1,422 @@
+
+JDA v5 provides type-trimmed channel managers, which provide only the setters that we know *for sure* can work on the given channel. Each of these managers can be found in [the `net.dv8tion.jda.api.managers.channel` package](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/managers/channel/package-summary.html). These all follow the same implement/extension hierarchy as the channels do and map 1:1.
+
+### Independant Stage Channel and News Channel Entities
+
+[`StageChannel`](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/concrete/StageChannel.html) and [`NewsChannel`](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/concrete/NewsChannel.html) were previously variants of `VoiceChannel` and `TextChannel`. These are now their own independant entities. This comes with a number of changes:
+
+- `ChannelAction#setNews` has been removed in favor of `ChannelAction.setType`
+- `ChannelManager#getType` was removed in favor of `ChannelManager#getChannel()#getType`
+- `ChannelManager#setNews` has been removed in favor of `ChannelManager#setType`
+- `TextChannel#crosspostMessage` was removed in favor of `NewsChannel#crosspostMessage`
+- `TextChannel#follow` has been removed in favor of `NewsChannel#follow`
+- `TextChannel#isNews` has been removed
+
+### Other Changes
+
+- `StoreChannel` has been removed
+- `PrivateChannel#getUser` is now nullable
+- `MessageChannel#getLatestMessageId` and `MessageChannel#getLatestMessageIdLong` no longer change to null if the message was deleted
+
+## Permission Changes
+
+A number of permissions have been renamed, with one permission being removed entirely.
+
+- `Permission.MANAGE_EMOTES` was renamed to `Permission.MANAGE_EMOTES_AND_STICKERS`
+- `Permission.MESSAGE_READ` has been removed in favor of `Permission.VIEW_CHANNEL`
+- `Permission.MESSAGE_WRITE` was renamed to `Permission.MESSAGE_SEND`
+- `Permission.USE_SLASH_COMMANDS` was renamed to `Permission.USE_APPLICATION_COMMANDS`
+- `Permission.USE_PUBLIC_THREADS` was renamed to `Permission.CREATE_PUBLIC_THREADS`
+- `Permission.USE_PRIVATE_THREADS` was renamed to `Permission.CREATE_PRIVATE_THREADS`
+
+## Event Changes
+
+There are several changes related to events.
+
+### Removal of Guild Context Message Events
+
+All message event types for guild messages have been removed. Examples of these are:
+
+- `Generic<Context>MessageEvent` (ex: `GenericGuildMessageEvent`)
+- `<Context>MessageReceivedEvent` (ex: `GuildMessageReceivedEvent`)
+- `<Context>MessageUpdateEvent` (ex: `GuildMessageUpdateEvent`)
+- `<Context>MessageDeleteEvent` (ex: `GuildMessageDeleteEvent`)
+- `<Context>MessageEmbedEvent` (ex: `GuildMessageEmbedEvent`)
+- `<Context>MessageReaction<X>Event` (ex: `GuildMessageReactionAddEvent`)
+- etc
```

Comment:
I don't think you need to include `etc` when these are non-comprehensive examples anyway